### DISCUSSION TITLE
runtime_tmpdir

### DISCUSSION THREAD

**Discussion by MickeMannen** [pyinstaller/pyinstaller#7547] [Help] [answered]
I am currently using pyinstaller version: 5.7 (will try with 5.9 after this message) on MacOS Ventura

I am trying to set the tmp dir for my application in the .spec file.
If i use an absolute path like "/Users/mike/tmp_folder" it works fine but if i try to change this so i can have the tmp folder in the users home directory like. "~/tmp_folder" it doesnt work. How can i add the path for runtime_tmpdir so it points to the users home folder?

//Micke

**Comment by rokm** [comment, marked as answer]
PyInstalle'rs bootloader does not perform environment variable expansion on your custom `runtime_tmpdir` string, so you cannot use environment variables, such as `~` (or `$HOME`). So you cannot set runtime dir to dynamically resolve your home directory, except perhaps by using default runtime_tmpdir and overriding `TMPDIR` in a wrapper script that launches your executable.

**Reply by MickeMannen** [reply]
Thank you!

I used to set the tmpdir in a start.sh script but based on some amazing fixes in osx 13.3 i can't use the start script anymore without annoying pop ups.

**Reply by paul-ngo** [reply]
Can we get this added to the official documentation regarding the use of the runtime_tmpdir argument? 

This is a great alternative that is working well for my use case. I spent way too much time trying to get the runtime_tmpdir to work with dynamic paths before finding this suggestion.

**Reply by rokm** [reply]
> Can we get this added to the official documentation regarding the use of the runtime_tmpdir argument?

@paul-ngo Does #8379 add what you had in mind?

**Reply by paul-ngo** [reply]
Yes - I added a comment for additional context on when this is useful. 

Thanks!
